### PR TITLE
ELPP-3096 integrate auth0

### PR DESCRIPTION
**Pull request by de-code**
The auth0 client id looks like it shouldn't be made public but it's sent to the browser anyway (and that is how the API is meant to be used).

### CODE REVIEW

File: salt/pillar/reviewer-suggestions.sls [hunk 1]
```diff
@@ -14,3 +14,17 @@ reviewer_suggestions:
         password: barpass
         host: 127.0.0.1
         port: 5432
+
+    auth:
+        auth0_domain: elife.auth0.com
+        auth0_client_id: s0mwbpPUuDBuYAAFMxYUT6J9xF6pn67O
+        valid_domains: elifesciences.org
```
Comment (giorgiosironi) [inline salt/pillar/reviewer-suggestions.sls]: I guess they check usage on domains so that the client id cannot be reused

**Review by giorgiosironi — APPROVED** [review]
Would need a builder-private addition for the pillars

File: salt/pillar/reviewer-suggestions.sls (hunk 1 above)
Comment (de-code) [reply salt/pillar/reviewer-suggestions.sls]: The `auth0_domain` is the API entry point for Auth0.

The `valid_domains` is actually something I added to only allow access with emails from a specific domain or a fixed list of emails. It is possible to do that via auth0 rules / hooks but it was simpler without auth0 being involved in that. Perhaps `valid_email_domains` would be a better name.

Comment (giorgiosironi) [reply salt/pillar/reviewer-suggestions.sls]: :+1:

**Review by giorgiosironi — APPROVED** [review]

**Pull request merged**